### ISSUE TITLE
[TW#24058] Missing typedef on esp_image_spi_freq_t

### ISSUE THREAD

**Issue by nkolban** [espressif/esp-idf#2136]
In the source header file called `components/bootloader_support/include/esp_image_format.h` there is a data type called `esp_image_spi_freq_t` which is an enum.  I believe that there is a slight error here.  I believe we are missing a `typedef` statement component at the start of the enum.

Today it reads:

```
enum {
    ESP_IMAGE_SPI_SPEED_40M,
    ESP_IMAGE_SPI_SPEED_26M,
    ESP_IMAGE_SPI_SPEED_20M,
    ESP_IMAGE_SPI_SPEED_80M = 0xF
} esp_image_spi_freq_t;
```

I believe it should read:

```
typedef enum {
    ESP_IMAGE_SPI_SPEED_40M,
    ESP_IMAGE_SPI_SPEED_26M,
    ESP_IMAGE_SPI_SPEED_20M,
    ESP_IMAGE_SPI_SPEED_80M = 0xF
} esp_image_spi_freq_t;
```

**Comment by markwj** [comment]
FYI: We are also seeing this. nkolban's suggested change works for us.

**Comment by ginkgm** [comment]
@FayeY this is fixed in 24e728669378db6723f71c1e5067fe452b080e4f, please close it.
